browser-extension/mv2/src/background/background.js

```diff
@@ -1087,31 +1111,74 @@ BG.Methods.onAppLoadedNotification = () => {
   EventActions.sendExtensionEvents();
 };
 
-BG.Methods.onContentScriptLoadedNotification = async (tabId) => {
-  const cachedAppliesRules = BG.Methods.getCachedAppliedRuleDetails(tabId);
+BG.Methods.onContentScriptLoadedNotification = async (tab, payload = {}) => {
+  if (payload.isIframe) {
+    return;
+  }
+
+  const cachedAppliesRules = BG.Methods.getCachedAppliedRuleDetails(tab.id);
 
   if (cachedAppliesRules?.length > 0) {
     chrome.tabs.sendMessage(
-      tabId,
+      tab.id,
       {
         action: RQ.CLIENT_MESSAGES.SYNC_APPLIED_RULES,
         appliedRuleDetails: cachedAppliesRules,
         isConsoleLoggerEnabled: await RQ.StorageService.getRecord(RQ.CONSOLE_LOGGER_ENABLED),
       },
-      () => window.tabService.removeData(tabId, "appliedRuleDetails")
+      () => window.tabService.removeData(tab.id, BG.TAB_SERVICE_DATA.APPLIED_RULE_DETAILS)
     );
   }
 
-  if (window.tabService.getData(tabId, "recordSession") === true) {
-    chrome.tabs.sendMessage(tabId, { action: RQ.CLIENT_MESSAGES.START_RECORDING, notify: true }, () => {
-      window.tabService.removeData(tabId, "recordSession");
-    });
+  let sessionRecordingData = window.tabService.getData(tab.id, BG.TAB_SERVICE_DATA.SESSION_RECORDING);
+
+  if (!sessionRecordingData) {
+    const sessionRecordingConfig = await BG.Methods.getSessionRecordingConfig(tab.url);
+
+    if (sessionRecordingConfig) {
+      sessionRecordingData = { config: sessionRecordingConfig };
+      window.tabService.setData(tab.id, BG.TAB_SERVICE_DATA.SESSION_RECORDING, sessionRecordingData);
+    }
+  }
+
+  if (sessionRecordingData) {
+    chrome.tabs.sendMessage(
+      tab.id,
+      {
+        action: RQ.CLIENT_MESSAGES.START_RECORDING,
+        payload: sessionRecordingData,
+      },
+      () => {
+        window.tabService.setData(tab.id, BG.TAB_SERVICE_DATA.SESSION_RECORDING, {
+          ...sessionRecordingData,
+          notify: false,
+          previousSession: null,
+        });
+      }
+    );
   }
 };
 
-BG.Methods.startRecordingOnUrl = (url) => {
+BG.Methods.startRecordingExplicitly = (tabId) => {
+  const sessionRecordingData = { explicit: true };
+
+  window.tabService.setData(tabId, BG.TAB_SERVICE_DATA.SESSION_RECORDING, sessionRecordingData);
+
+  chrome.tabs.sendMessage(tabId, {
+    action: RQ.CLIENT_MESSAGES.START_RECORDING,
+    payload: sessionRecordingData,
+  });
+};
+
+BG.Methods.launchUrlAndStartRecording = (url) => {
   chrome.tabs.create({ url }, (tab) => {
-    window.tabService.setData(tab.id, "recordSession", true);
+    window.tabService.setData(tab.id, BG.TAB_SERVICE_DATA.SESSION_RECORDING, { notifyOnStart: true });
```

Comment:
```suggestion
    window.tabService.setData(tab.id, BG.TAB_SERVICE_DATA.SESSION_RECORDING, { notify: true });
```
We read `notify` in `RQ.SessionRecorder.startRecording`